### PR TITLE
Update contact webrole changes

### PR DESCRIPTION
**Pull request by burzynskijakub**

### CODE REVIEW

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investments.Organisation/Services/ContactService.cs [hunk 1]
```diff
@@ -185,6 +182,44 @@ public async Task UpdateContactWebrole(IOrganizationServiceAsync2 service, strin
...
+        {
+            var contactExternalFilter = "<condition attribute=\"invln_externalid\" operator=\"in\">";
+            foreach (var contactExternal in contactExternalId)
+            {
+                contactExternalFilter += $"<value>{contactExternal}</value>";
+            }
+            contactExternalFilter += "</condition>";
+
+            var contactWebroles = _webRoleRepository.GetWebrolesForPassedContacts(service, contactExternalFilter, organisationGuid);
+            if (contactWebroles.Any())
```
Comment (Antek012) [inline src/HE.Investments.Organisation/Services/ContactService.cs:198]: if is not needed

File: src/HE.Investments.Organisation/Services/ContactService.cs [hunk 2]
```diff
@@ -185,6 +182,44 @@ public async Task UpdateContactWebrole(IOrganizationServiceAsync2 service, strin
...
+            {
+                foreach (var contactWebrole in contactWebroles)
+                {
+                    contactRolesList.Add(new ContactRolesDto()
```
Comment (Antek012) [inline src/HE.Investments.Organisation/Services/ContactService.cs:202]: please create private factory method to improve readability

File: src/HE.Investments.Organisation/Services/ContactService.cs [hunk 3]
```diff
@@ -185,6 +182,44 @@ public async Task UpdateContactWebrole(IOrganizationServiceAsync2 service, strin
...
+                    {
+                        contactRoles = new List<ContactRoleDto>()
+                        {
+                            new ContactRoleDto()
+                            {
+                                accountId = contactWebrole.Contains("invln_accountid") && contactWebrole["invln_accountid"] != null ? ((EntityReference)contactWebrole["invln_accountid"]).Id : Guid.Empty,
+                                accountName = contactWebrole.Contains("acc.name") && contactWebrole["acc.name"] != null ? contactWebrole.GetAttributeValue<AliasedValue>("acc.name").Value.ToString() : string.Empty,
+                                permissionLevel = contactWebrole.Contains("ppl.invln_name") && contactWebrole["ppl.invln_name"] != null ? contactWebrole.GetAttributeValue<AliasedValue>("ppl.invln_name").Value.ToString() : null,
+                                webRoleName = contactWebrole.Contains("wr.invln_name") ? contactWebrole.GetAttributeValue<AliasedValue>("wr.invln_name").Value.ToString() : string.Empty,
```
Comment (Antek012) [inline src/HE.Investments.Organisation/Services/ContactService.cs:211]: code
`contactWebrole.Contains("wr.invln_name") ? contactWebrole.GetAttributeValue<AliasedValue>("wr.invln_name").Value.ToString() : string.Empty`
is repeated in this class in many places
Can you create method GetValueOrDefault and capsulate parsing logic?

File: src/HE.Investments.Organisation/Services/ContactService.cs [hunk 4]
```diff
@@ -98,76 +99,72 @@ public async Task UpdateUserProfile(IOrganizationServiceAsync2 service, string c
...
-    public async Task<List<ContactDto>> GetAllOrganisationContactsForPortal(IOrganizationServiceAsync2 service, string organizationNumber, string? portalType = null)
+    public Task<List<ContactDto>> GetAllOrganisationContactsForPortal(IOrganizationServiceAsync2 service, Guid organisationGuid, int? portalType = null)
     {
-        var organisation = await _organizationRepository.SearchForOrganizationsByOrganizationId(service, organizationNumber);
-        organisation ??= _organizationRepository.GetOrganizationViaCompanyHouseNumber(service, organizationNumber);
         var contactList = new List<ContactDto>();
-        if (organisation != null)
+        var portalTypeFilter = GeneratePortalTypeFilter(portalType);
+        var contacts = _contactRepository.GetContactsForOrganisation(service, organisationGuid, portalTypeFilter);
+        if (contacts.Any())
```
Comment (Antek012) [inline src/HE.Investments.Organisation/Services/ContactService.cs:148]: if not needed

**Review by Antek012 — APPROVED** [review]

**Pull request merged**